test/e2e/traceflow_test.go

```diff
@@ -460,7 +481,12 @@ func TestTraceflow(t *testing.T) {
 func (data *TestData) waitForTraceflow(name string, phase v1alpha1.TraceflowPhase) (*v1alpha1.Traceflow, error) {
 	var tf *v1alpha1.Traceflow
 	var err error
-	if err = wait.PollImmediate(1*time.Second, 15*time.Second, func() (bool, error) {
+	timeout := 15 * time.Second
+	if phase == v1alpha1.Failed {
+		// 2 minutes timeout + 1 minute check interval
+		timeout = 3 * time.Minute
+	}
+	if err = wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
```

Comment:
do you think it's worth increase e2e test runtime that much? can this be tested as a unit test in the controller package, leveraging a configurable timeout value? like we do for certificate rotation testing

Reply:
What is the current expected time for e2e tests?
For unit tests, I need to take a further look to understand.

Reply:
I believe it's around 20-25 minutes. But it's very easy to keep increasing that duration by adding "slow" tests. We used to run all these Traceflow tests in parallel but for some reason it is no longer the case.

Reply:
I did not figure out how to add unit tests. Tried "client/clientset/versioned/fake", but Traceflow Controller syncTracflow will ignore the any ADD event, as traceflowLister.Get() will not return any Traceflow object.

@tnqn : do you think any way to add unit tests?

Reply:
If you create the informerFactory with the fakeClient, the Lister should get the objects.
https://github.com/vmware-tanzu/antrea/blob/218ad95167842f5e9393ae47dec3ee607fd477c3/pkg/controller/networkpolicy/networkpolicy_controller_test.go#L81-L83

Reply:
That is what I tried, but still traceflowLister.Get() cannot find the object. Pushed my test code. Could you help check?

Reply:
informerFactory.Start() works.

Reply:
Added unit tests. Please check. @antoninbas